### PR TITLE
Incorrect nodePort chosen for SSL healthcheck

### LINKED ISSUE #98826: Incorrect nodePort chosen for SSL healthcheck
**What happened**:
Having http and https ports on Service object in conjunction with "service.beta.kubernetes.io/aws-load-balancer-backend-protocol: https" causes healthcheck to send SSL traffic to http port.

**What you expected to happen**:
Either use TCP protocol for tcp listener in case we have backend-protocol set to "https" or choose correct nodePort.

**How to reproduce it (as minimally and precisely as possible)**:
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags: Name=ingress-public
    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: https
    service.beta.kubernetes.io/aws-load-balancer-connection-draining-enabled: "true"
    service.beta.kubernetes.io/aws-load-balancer-connection-draining-timeout: "30"
    service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout: "60"
    service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold: "2"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval: "15"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout: "5"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold: "3"
    service.beta.kubernetes.io/aws-load-balancer-internal: "false"
    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:<ARN>
    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: https
  labels:
    app.kubernetes.io/component: controller
    app.kubernetes.io/instance: ingress-public
    app.kubernetes.io/name: ingress-public
  name: ingress-public
  namespace: ingress-public
spec:
  externalTrafficPolicy: Cluster
  ports:
  - name: http
    nodePort: 31544
    port: 80
    protocol: TCP
    targetPort: http
  - name: https
    nodePort: 32591
    port: 443
    protocol: TCP
    targetPort: https
  selector:
    app.kubernetes.io/component: controller
    app.kubernetes.io/instance: ingress-public
    app.kubernetes.io/name: ingress-public
  sessionAffinity: None
  type: LoadBalancer
```
Create service described above and it will configure healthcheck by setting protocol to SSL and will use nodePort from the first listener(HTTP). It will cause healthcheck traffic to use SSL and send it to HTTP port and fail(will mark instance OutOfService).
Healthcheck will be set as: SSL:31544 in our case.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): master
- Cloud provider or hardware configuration: AWS(EKS)
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by sanicheev**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #98826 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

**Comment by sanicheev** [comment]
Can anyone have a look at this PR please?

**Comment by neolit123** [comment]
> Can anyone have a look at this PR please?

you can try asking in #sig-cloud-provider on k8s slack for guidance.

**Comment by kishorj** [comment]
k8s 1.20 and later support the service annotation `service.beta.kubernetes.io/aws-load-balancer-healthcheck-port` to create custom health check port. I recommend using the annotation instead of this fix.

**Comment by sanicheev** [comment]
> k8s 1.20 and later support the service annotation `service.beta.kubernetes.io/aws-load-balancer-healthcheck-port` to create custom health check port. I recommend using the annotation instead of this fix.

What if you have 20+ loadbalancers?
Will you be checking the port everytime you need to create a new one?
This is not a proper way of doing it.

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 1]
```diff
@@ -4144,15 +4160,18 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
...
 		var hcProtocol string
 		if annotationProtocol == "https" || annotationProtocol == "ssl" {
 			hcProtocol = "SSL"
 		} else {
 			hcProtocol = "TCP"
 		}
-		// there must be no path on TCP health check
-		err = c.ensureLoadBalancerHealthCheck(loadBalancer, hcProtocol, tcpHealthCheckPort, "", annotations)
+		// there must be no path on TCP/SSL health check
+		if sslHealthCheckPort == 0 {
```
[the remark this answers is not in the document]
Comment (sanicheev) [reply staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:4170]: i can change that, but in current implementation its easier to control return statement based on that condition

**Comment by sanicheev** [comment]
Will not change lifecycle labels again. Let there be bugs in k8s.

**Pull request closed without merging**